Add Funding page navbar link

Kobzol opened the pull request:
As a part of the effort to promote it (https://rust-lang.zulipchat.com/#narrow/channel/392734-council/topic/Publicizing.20our.20funding.20page/with/560030032, https://github.com/rust-lang/blog.rust-lang.org/pull/1751).

<img width="903" height="91" alt="image" src="https://github.com/user-attachments/assets/d54ac446-68a0-4dfb-a4cd-89b652f9757a" />

We should probably translate it into all current languages, since an untranslated navbar item will probably look quite weird?

Marking as a draft, to sync it with the blog post and other changes.

Kobzol committed:
Add Funding page navbar link

Manishearth reviewed (approved):
I'm fine with this.

*[View changes since this review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/www.rust-lang.org/2245/84e18218f1265c1f03f56f91905a98b84098c506..88d3e3b6e54b55ade959f4bdb4540720212f4733)*

Manishearth commented:
> Since it is in the navbar, I think that we should translate this into other languages. I know that there is a freeze on new translations, but having untranslated text in the navbar feels.. meh. Do you agree?

Yes. We can ask around for translations + potentially use some machine translations. It's easier for single words since you can look it up in a dictionary and be sure it's the right sense of the word.

lqd committed:
add some "Funding" translations

These were double-checked, improved, or authored by project members:
- jieyouxu
- oli-obk
- Miguel Ojeda
- apiraino
- TaKO8Ki
- ShoyuVanilla
- petrochenkov
- lqd

Kobzol committed:
Merge pull request #2 from lqd/funding-translations

add some "Funding" translations

Kobzol commented:
We talked about this with Rémy :laughing: ... yes, in retrospect I think that donate is in fact a better name for the link :man_facepalming:
